simple/msg_pingpong.c

```diff
@@ -543,77 +535,31 @@ static int run(void)
 	return ret;
 }
 
-void print_usage(char *name)
-{
-	fprintf(stderr, "Usage:\n");
-	fprintf(stderr, "  %s [OPTIONS]\t\tstart ping server\n", name);
-	fprintf(stderr, "  %s [OPTIONS] <host>\tpong given host\n", name);
-
-	fprintf(stderr, "\nOptions:\n");
-	fprintf(stderr, "  -n\tdomain name\n");
-	fprintf(stderr, "  -p\tport number\n");
-	fprintf(stderr, "  -s\tsource address\n");
-	fprintf(stderr, "  -I\tnumber of iterations\n");
-	fprintf(stderr, "  -S\tspecific transfer size or 'all'\n");
-	fprintf(stderr, "  -m\tmachine readable output\n");
-	fprintf(stderr, "  -i\tprint hints structure and exit\n");
-	fprintf(stderr, "  -v\tdisplay versions and exit\n");
-	fprintf(stderr, "  -h\tdisplay this help output\n");
-
-	return;
-}
-
 int main(int argc, char **argv)
 {
 	int op, ret;
-	int prhints = 0;
 
-	while ((op = getopt(argc, argv, "n:p:s:I:S:mivh")) != -1) {
+	/* default options for test */
+	opts.iterations = 1000;
+	opts.transfer_size = 1024;
+	opts.port = "9228";
+	opts.argc = argc;
+	opts.argv = argv;
+
+	while ((op = getopt(argc, argv, "vhg:" CSOPTS INFOOPTS)) != -1) {
 		switch (op) {
-		case 'n':
-			domain_hints.name = optarg;
-			break;
-		case 'p':
-			port = optarg;
-			break;
-		case 's':
-			ret = getaddr(optarg, NULL,
-				      (struct sockaddr **) &hints.src_addr,
-				      (socklen_t *) &hints.src_addrlen);
-
-			src_addr = optarg;
-			if (ret) {
-				FI_DEBUG("source address error %s\n", gai_strerror(ret));
-				return EXIT_FAILURE;
-			}
-			break;
-		case 'I':
-			custom = 1;
-			iterations = atoi(optarg);
-			break;
-		case 'S':
-			if (!strncasecmp("all", optarg, 3)) {
-				size_option = 1;
-			} else {
-				custom = 1;
-				transfer_size = atoi(optarg);
-			}
-			break;
-		case 'm':
-			machr = 1;
-			g_argc = argc;
-			g_argv = argv;
-			break;
-		case 'i':
-			prhints = 1;
-			break;
 		case 'v':
-			printf("%s: 0.0.1\n", argv[0]);
-			printf("%s: %s\n", PACKAGE_NAME, PACKAGE_VERSION);
+			printf("%s: %s\n", argv[0], PACKAGE_VERSION);
+			printf("libfabric: %s\n", fi_tostr("1", FI_TYPE_VERSION));
+			printf("libfabric api: %d.%d\n", FI_MAJOR(FT_FIVERSION), FI_MINOR(FT_FIVERSION));
 			return EXIT_SUCCESS;
-		case 'h':
 		default:
-			print_usage(argv[0]);
+			ft_parseinfo(op, optarg, &hints);
+			ft_parsecsopts(op, optarg, &opts);
```

Comment:
We are not printing usage for unknown options here. May be use return values from the parse functions to check if we received any unknown option?